### PR TITLE
Features/delly call recipe

### PR DESCRIPTION
**Pull request by northwestwitch**
Had to create a module to remove contigs from a list of contigs because the recipe was using remove_array_element from main mip executable. I named this new module "Contigshandler" and placed it under MIP::Program::Interval. After your review of this pull request I'll modify the rest of the code in mip to use this new module instead of "remove_array_element" in mip (which can be ultimately removed).
Tested recipe on rasta!

### CODE REVIEW

**Comment by northwestwitch** [comment]
Gee I just realized that my explanation is more complicated than the code itself! :)

**Comment by northwestwitch** [comment]
Wait a moment, while I was working on Delly reformat I stumbled upon MIP::Delete::List qw{ delete_contig_elements }. So there was no need for me to create an extra module to do the same, I'll use this instead!

File: lib/MIP/Program/Interval/Contigshandler.pm [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package MIP::Program::Interval::Contigshandler;
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Interval/Contigshandler.pm]: I think you can replace this with: MIP::Delete::List qw{ delete_contig_elements }. I noticed that you already noticed, 😄

File: lib/MIP/Recipes/Analysis/Delly_call.pm [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+
+    ## Add merged infile name prefix after merging all BAM files per sample_id
+    my $merged_infile_prefix = get_merged_infile_prefix(
+        {
+            file_info_href => $file_info_href,
+            sample_id      => $sample_id,
+        }
+    );
+
+    ## Files
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply lib/MIP/Recipes/Analysis/Delly_call.pm]: Ehm..

File: lib/MIP/Recipes/Analysis/Delly_call.pm [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+            file_suffix => $parameter_href->{$mip_program_name}{outfile_suffix},
+        }
+    );
+
+    ### Update contigs
+    ## Removes an element from array and return new array while leaving orginal elements_ref untouched
+    my @contigs = delete_contig_elements(
+        {
+            elements_ref       => \@{ $file_info_href->{contigs_size_ordered} },
+            remove_contigs_ref => [qw{ MT M }],
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply lib/MIP/Recipes/Analysis/Delly_call.pm]: I used this code
my @contigs = delete_contig_elements(
        {
            elements_ref       => \@{ $file_info_href->{contigs_size_ordered} },
            remove_contigs_ref => [qw{ MT M Y }],
        }
    );

It's removing Mt and Y, same output as in master

File: lib/MIP/Recipes/Analysis/Delly_call.pm [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+            parameter_href => $parameter_href,
+            suffix_key     => q{variant_file_suffix},
+            job_id_chain   => $job_id_chain,
+            file_suffix => $parameter_href->{$mip_program_name}{outfile_suffix},
+        }
+    );
+
+    ### Update contigs
+    ## Removes an element from array and return new array while leaving orginal elements_ref untouched
+# Skip contig Y along with MT throughout since sometimes there are no variants particularly for INS
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_call.pm]: Indent comment

File: lib/MIP/Recipes/Analysis/Delly_call.pm [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+                    $insample_directory,
+                    $infile_prefix . substr( $infile_suffix, 0, 2 ) . $ASTERISK
+                ),
+                outfile_path => $active_parameter_href->{temp_directory}
+            }
+        );
+    }
+
+    if (
+        (
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_call.pm]: Remove line

File: lib/MIP/Recipes/Analysis/Delly_call.pm [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+            any { /DEL|DUP|INS|INV/ }
+            @{ $active_parameter_href->{delly_types} }
+        )
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_call.pm]: Remove line

File: lib/MIP/Recipes/Analysis/Delly_call.pm [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+                infile             => $infile_prefix,
+                indirectory        => $insample_directory,
+                file_ending    => substr( $infile_suffix, 0, 2 ) . $ASTERISK,
+                temp_directory => $temp_directory,
+            }
+        );
+        say {$FILEHANDLE} q{wait}, $NEWLINE;
+    }
+
+    ## delly
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_call.pm]: delly -> Delly

**Comment by henrikstranneheim** [comment]
👍

**Comment by northwestwitch** [comment]
I'm working on Delly reformat now!

Chiara

**Comment by henrikstranneheim** [comment]
Perfect!

**Pull request merged**